scripts/testBuiltTypes.mjs

```diff
@@ -1,38 +1 @@
-import glob from 'fast-glob';
-import fse from 'fs-extra';
-import path from 'path';
-import { getWorkspaceRoot } from './utils.mjs';
-
-async function main() {
-  const workspaceRoot = getWorkspaceRoot();
-
-  const declarationFiles = await glob('**/build/**/*.d.ts', {
-    absolute: true,
-    cwd: workspaceRoot,
-    ignore: ['node_modules'],
-    followSymbolicLinks: false,
-  });
-
-  await Promise.all(
-    declarationFiles.map(async (declarationFilePath) => {
-      const declarationFile = await fse.readFile(declarationFilePath, { encoding: 'utf8' });
-      // find occurrences of e.g. `import("../../mui-*/src/...")`
-      const typeImportsRelativeToWorkspace = declarationFile.match(/import\(("|')(\.\.\/)+mui/g);
-
-      if (typeImportsRelativeToWorkspace !== null) {
-        console.error(
-          // readable path for CI while making it clickable locally
-          `${path.relative(process.cwd(), declarationFilePath)} possibly imports types ${
-            typeImportsRelativeToWorkspace.length
-          } times that are unreachable once published.`,
-        );
-        process.exitCode = 1;
-      }
-    }),
-  );
-}
-
-main().catch((error) => {
-  console.error(error);
-  process.exit(1);
-});
+import '@mui/monorepo/scripts/testBuiltTypes.mjs';
```

Comment:
Can keep it for now, but looking at the script, I assume this is unnecessary if your declarations are built from typescript files.

Reply:
> Actually, it's looking for mui-* segments in the paths
True. I wonder if it was even working as intended.